Client/Frontend/Settings/SettingsViewController.swift

```diff
@@ -196,6 +197,54 @@ class SettingsViewController: TableViewController {
                     self.navigationController?.pushViewController(clearPrivateData, animated: true)
                 },
                 accessory: .disclosureIndicator
+            ),
+            Row(
+                text: Strings.Block_all_cookies,
+                accessory: .switchToggle(
+                    value: Preferences.Privacy.blockAllCookies.value, { [unowned self] in
+                        func perform(with status: Bool) {
+                            //Lock/Unlock Cookie Folder
+                            let success = FileManager.default.lockFolders([
+                                (.cookie, status),
+                                (.webSiteData, status)
+                                ])
+                            if success {
+                                Preferences.Privacy.blockAllCookies.value = status
+                            } else {
+                                //Revert the changes. Not handling success here to avoid a loop.
+                                _ = FileManager.default.lockFolders([
+                                    (.cookie, !status),
+                                    (.webSiteData, !status)
+                                    ])
+                                self.undoToggleEvent(section: 1, row: 1)
+                                
+                                // TODO: Throw Alert to user to try again?
+                                let alert = UIAlertController(title: nil, message: Strings.Block_all_cookies_failed_alert_msg, preferredStyle: .alert)
+                                let okAction = UIAlertAction(title: Strings.OKString, style: .default)
+                                alert.addAction(okAction)
+                                self.present(alert, animated: true)
+                            }
+                        }
+                        if $0 {
+                            let status = $0
+                            // THROW ALERT to inform user of the setting
+                            let alert = UIAlertController(title: Strings.Block_all_cookies_alert_title, message: Strings.Block_all_cookies_alert_info, preferredStyle: .alert)
+                            let okAction = UIAlertAction(title: Strings.OKString, style: .default, handler: { (action) in
+                                perform(with: status)
+                            })
+                            alert.addAction(okAction)
+                            
+                            let cancelAction = UIAlertAction(title: Strings.CancelButtonTitle, style: .cancel, handler: { (action) in
+                                self.undoToggleEvent(section: 1, row: 1)
```

Comment:
Probably not a good idea to rely on hardcoded section/row numbers. This might break when we add Sync (can't remember currently where it goes in settings off the top of my head) or other settings. Maybe stick to the section/row uuid to obtain a section/row index?

Reply:
Agreed, changed to as suggested :)